devtools/gradle/src/test/java/io/quarkus/gradle/GradleBuildFileTest.java

```diff
@@ -0,0 +1,88 @@
+package io.quarkus.gradle;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.maven.model.Dependency;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.cli.commands.writer.FileProjectWriter;
+
+class GradleBuildFileTest {
+
+    private static GradleBuildFileFromConnector buildFile;
+
+    @BeforeAll
+    public static void beforeAll() throws IOException, URISyntaxException {
+        URL url = GradleBuildFileTest.class.getClassLoader().getResource("gradle-project");
+        URI uri = url.toURI();
+        Path gradleProjectPath;
+        if (uri.getScheme().equals("jar")) {
+            FileSystem fileSystem = FileSystems.newFileSystem(uri, Collections.<String, Object> emptyMap());
```

Comment:
Normally this should be a directory, right? When would that be found in a JAR? FileSystem should be closed then.

Reply:
you are right